### PR TITLE
Refactor audit component

### PR DESCRIPTION
**Pull request by jonahkagan**
**Description**

Separate the `Audit` component into different components for diff audit types/user types.

**Testing**

Existing tests cover.

**Progress**

Ready

### CODE REVIEW

**Comment by MorganLove** [comment]
@jonahkagan Interesting that the backend tests are failing. Huh.

But yeah, this looks like a great direction to go in!

**Comment by jonahkagan** [comment]
Weird, I couldn't repro that backend test failure at all. I tried so many times on that commit, and on different branches. If you see it again, let me know.

File: arlo-client/src/components/Audit/index.test.tsx [hunk 1]
```diff
@@ -207,9 +207,15 @@ describe('RiskLimitingAuditForm', () => {
 })
 
 describe('AA setup flow', () => {
+  // MultiJurisdictionAudit will only be rendered once the user is logged in, so
+  // we simulate that.
+  const MultiJurisdictionAuditWithAuth: React.FC = () => {
+    const { isAuthenticated } = useContext(AuthDataContext)
+    return isAuthenticated ? <MultiJurisdictionAudit /> : null
```
Comment (benadida) [inline arlo-client/src/components/Audit/index.test.tsx:214]: is `null` the right alternate return value here, or should it be an empty react fragment?

File: arlo-client/src/components/CreateAudit.test.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const checkAndToastMock: jest.SpyInstance<
 checkAndToastMock.mockReturnValue(false)
 
 const routeProps: RouteComponentProps<ICreateAuditParams> = routerTestProps(
-  '/election/:electionId',
+  '/audit/:electionId',
```
Comment (benadida) [inline arlo-client/src/components/CreateAudit.test.tsx:23]: wait, did i miss a URL change? I'm not seeing it

Comment (benadida) [reply arlo-client/src/components/CreateAudit.test.tsx:23]: or is this just for testing? If so it's a tad confusing.

File: arlo-client/src/components/CreateAudit.tsx [hunk 3]
```diff
@@ -70,7 +70,11 @@ const CreateAudit = ({ history }: RouteComponentProps<ICreateAuditParams>) => {
         return
       }
       const { electionId } = response
-      history.push(`/election/${electionId}/setup`)
+      if (isAuthenticated) {
+        history.push(`/election/${electionId}/setup`)
+      } else {
+        history.push(`/audit/${electionId}/setup`)
```
Comment (benadida) [inline arlo-client/src/components/CreateAudit.tsx:76]: for the life of me I can't find this `/audit/` route... where is it?

File: arlo-client/src/components/Audit/index.test.tsx (hunk 1 above)
Comment (jonahkagan) [reply arlo-client/src/components/Audit/index.test.tsx:214]: https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

File: arlo-client/src/components/CreateAudit.test.tsx (hunk 2 above)
Comment (jonahkagan) [reply arlo-client/src/components/CreateAudit.test.tsx:23]: https://github.com/votingworks/arlo/pull/464/files#diff-c4fc915a5c89a9458ba5320c7f2be3f8R72-R73

the single-jurisdiction flow is going under `/audit` so we can use routing to know which flow we're supposed to render

**Pull request merged**